iroha-lib/examples/DomainAssetCreation.cpp

```diff
@@ -0,0 +1,251 @@
+#include <iostream>
+#include <string>
+#include <cassert>
+
+#include "model/Tx.hpp"
+#include "model/Query.hpp"
+#include "model/utils/Utils.h"
+
+
+constexpr const char* LINE = "-----------";
+
+
+iroha_lib::Query generateQueryBase(const std::string& account_name, const std::string& key_path);
+iroha::protocol::Query generateGetAccountAssetsQuery(const std::string& account_name, const std::string& key_path);
+iroha::protocol::Query generateGetAccountTransactionsQuery(const std::string& account_name, const std::string& key_path);
+iroha::protocol::Query generateGetAccountQuery(const std::string& account_name, const std::string& key_path);
+
+template<typename Tx>
+void sendTransaction(Tx& tx, const std::string& peer_ip, uint16_t torii_port);
+
+iroha::protocol::Transaction generateTransactionWhichCreatesDomainAndAsset(
+        const std::string& account_name,
+        const std::string& key_path,
+        const std::string& domain_id,
+        const std::string& user_default_role,
+        const std::string& asset_name);
+iroha::protocol::Transaction generateTransactionWhichAddsAssetQuantiti(
+        const std::string& account_name,
+        const std::string& key_path,
+        const std::string& assetIdWithDomain,
+        const std::string& assetAmount);
+
+void printAccountAssets(const std::string& account_name, const std::string& key_path, const std::string& peer_ip, uint16_t torii_port);
+void printAccount(const std::string& account_name, const std::string& key_path, const std::string& peer_ip, uint16_t torii_port);
+
+void run(const std::string& key_path);
+
+
+int main(int argc, char** argv)
+{
+    if (argc > 1 && argc < 3) {
+        run(argv[1]);
+    } else {
+        std::cout << "Usage: " << argv[0] << " key_path" << std::endl;
+    }
+}
```

Comment:
Might be a good idea to include another argument processing library here. Or maybe extract the common logic. IMO we want to set a good standard for mostly students.

Reply:
Do You mean to use: https://www.boost.org/doc/libs/1_81_0/doc/html/program_options/tutorial.html#id-1.3.30.4.3
? Should all those arguments be read from CMD:
```
    const auto adminAccountName = "admin@test";
    const auto peer_ip = "127.0.0.1";
    const uint16_t torii_port = 50051;
    const auto user_default_role = "user";

    const std::string assetName = "assetnamesamplev4";
    const std::string assetDomain = "domainsamplev4";
```

Reply:
Boost is a good starting point.

Reply:
I tried with boost::program_options, but after some time I noticed that in dependency list are some boost libraries expect for boost::program_options. That is why I decided to use command line library which is widely used by Iroha - gflags. It was first time I used the library, but probably I did what You wanted me to.